source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -125,6 +125,10 @@ DnsResolverImpl::AresOptions DnsResolverImpl::defaultAresOptions() {
     options.optmask_ |= ARES_OPT_NOROTATE;
   }
 
+  // Disable query cache by default.
+  options.optmask_ |= ARES_OPT_QUERY_CACHE;
+  options.options_.qcache_max_ttl = 0;
```

Comment:
Hm, this seems a bit beyond the scope of the PR description. Can you say more about why this is needed?